TestIntMem.java

```diff
@@ -0,0 +1,140 @@
+import java.util.Arrays;
+import java.util.Random;
+import java.lang.reflect.Method;
+import java.util.*;
+
+public class TestIntMem {
+
+    static int[] a;
+    static int[] b;
+    static int[] c;
+
+    public static int testKernel(int a, int b) {
+        int ans;
+        int k = Integer.rotateLeft(-a, 5);
+        int l = Integer.rotateRight(-b, 6);
+        int m = Integer.rotateLeft(a, Math.abs(Integer.bitCount(l) % 7));
+        ans = (a ^ b) + Math.max((a >> Math.min(2, k)), ((b >>> Math.min(3, l)) << 2));
+        return ans;
+    }
+
+    public static int testTzcnt(int x, int i) {
+        return  Integer.numberOfTrailingZeros(a[i]);
+    }
+
+    public static int testLzcnt(int x, int i) {
+        return  Integer.numberOfLeadingZeros(b[i]);
+    }
+
+    public static int testPopcnt(int x, int i) {
+        return  Integer.bitCount(a[i]);
+    }
+
+    public static int testRor(int a, int b) {
+        return  Integer.rotateRight(a, 8);
+    }
+
+    public static int testRol(int a, int b) {
+        return  Integer.rotateLeft(b, 7); // getting mapped to Ror 25
+    }
+
+
+    public static int testShrVar(int a, int b) {
+        return  a >>> (b >>> 28) ; // generates shrxl UseBMI2Instructions
+    }
+
+    public static int testShr(int a, int b) {
+        return  a >>> 6 ;
+    }
+
+    public static int testSal(int a, int b) {
+        return  a << 5 ;
+    }
+
+    public static int testSar(int a, int b) {
+        return  b >> 4 ;
+    }
+
+    public static int testDec(int a, int b) {
+        return  b - 1 ;
+    }
+
+    public static int testInc(int a, int b) {
+        return  a + 1 ;
+    }
+
+    public static int testNeg(int a, int b) {
+        return  -b;
+    }
+
+    public static int testXor(int a, int b) {
+        return a ^ b;
+    }
+
+    public static int testOr(int a, int b) {
+        return a | b;
+    }
+
+    public static int testAnd(int a, int b) {
+        return a & b;
+    }
+
+    public static int testCount(int a, int b) {
+        return Integer.bitCount(a) + Integer.numberOfLeadingZeros(b) + Integer.numberOfTrailingZeros(a-b);
+    }
+
+    public static int testMul(int a, int b) {
```

Comment:
Since these tests are specifically crafted for APX NDD support, it makes sense to check for the emission of APX instruction in JIT code using the IR framework.

You can add IR match rules for CompilePhase.FINAL_CODE.

Reply:
Test automation is the end goal. Currently, the generated code dump is being manually greped for the expected instruction. Once all the instructions are enabled, the next step is automation.